### PR TITLE
feat: add useToggleOrSet for convenience

### PR DESCRIPTION
**Pull request by BUPTlhuanyu**

### CODE REVIEW

**Comment by Justineo** [comment]
Why not just add a `force` argument to the `useToggle` hook?

**Comment by otakustay** [comment]
We can add force parameter to both useBoolean and useToggle

**Comment by BUPTlhuanyu** [comment]
maybe the useToggle and useSwitch can be replaced by useToggle + `force` parameter.

**Comment by otakustay** [comment]
I prefer this:

```js
const [value, {toggle}] = useBoolean();
toggle(true /false);

const [value, on, off, toggle] = uswSwitch();
toggle(true / false);

const [value, toggle] = useToggle();
toggle(true / false);
```

**Comment by BUPTlhuanyu** [comment]
@otakustay all right, i gonna fix it

File: packages/boolean/src/index.ts [hunk 1]
```diff
@@ -7,8 +7,8 @@ const reducers = {
     off() {
         return false;
     },
-    toggle(value: boolean) {
-        return !value;
+    toggle(prevState: boolean, arg: any) {
```
Comment (otakustay) [inline packages/boolean/src/index.ts:10]: `arg` should be boolean, strong typed hooks are always better

**Review by otakustay — CHANGES_REQUESTED** [review]

File: packages/boolean/src/index.ts (hunk 1 above)
Comment (BUPTlhuanyu) [reply packages/boolean/src/index.ts:10]: The type will be incorrect when i use the function as a event handler or some other callback，e.g.
```
<div on-click = { useToglle }></on-click>
<custom-comp on-selected= { useToggle }></custom-comp>
```

Comment (BUPTlhuanyu) [reply packages/boolean/src/index.ts]: 👌

**Pull request merged**